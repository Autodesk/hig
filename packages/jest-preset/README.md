# @hig/jest-preset

Jest preset for HIG packages.

## Usage

_Example `package.json`:_

```json
{
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
```
