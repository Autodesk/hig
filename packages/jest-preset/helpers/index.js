/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
