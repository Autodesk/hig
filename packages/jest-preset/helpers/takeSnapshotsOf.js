const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases, rendererOptions) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const vDom = React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
