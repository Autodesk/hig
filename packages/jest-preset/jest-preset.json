{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
