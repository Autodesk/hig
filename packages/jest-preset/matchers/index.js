require("./toBeConstants");
require("./toImplementHIGInterfaceOf");
require("./toHavePropTypesAndDocgenInfo");
