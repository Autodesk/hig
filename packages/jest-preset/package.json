{
  "private": true,
  "name": "@hig/jest-preset",
  "version": "0.1.0",
  "devDependencies": {
    "babel-core": "^6.26.3",
    "babel-jest": "^23.4.0",
    "enzyme": "^2.9.1",
    "jest": "^21.2.1",
    "jest-enzyme": "^3.8.3",
    "jest-extended": "^0.6.0",
    "raf": "^3.4.0",
    "react-test-renderer": "^15.6.1"
  },
  "eslintConfig": {
    "extends": "@hig"
  },
  "jest": {
    "preset": "@hig/jest-preset"
  }
}
