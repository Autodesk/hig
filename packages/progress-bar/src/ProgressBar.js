import React, { Component } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import "./progress-bar.scss";

/**
 * @param {Number} percent, an integer or float
 * @returns {Integer}
 */
export const renderedBarWidth = percent => {
  const percentageWidth = parseInt(percent, 10);
  if (!percentageWidth) {
    return null;
  }

  if (percentageWidth >= 100) {
    return 101;
  }
  return percentageWidth;
};

export default class ProgressBar extends Component {
  static propTypes = {
    /**
     * A number from 0 to 100 representing the percent the delayed operation has completed
     */
    percentComplete: PropTypes.number
  };

  fillBarStyles() {
    const barWidth = renderedBarWidth(this.props.percentComplete);
    return barWidth ? { width: `${barWidth}%` } : {};
  }

  render() {
    const { percentComplete } = this.props;
    const progressBarClasses = cx("hig__progress-bar", {
      "hig__progress-bar--determinate": percentComplete
    });

    return (
      <div
        className={progressBarClasses}
        role="progressbar"
        aria-valuemin="0"
        aria-valuemax="100"
        aria-valuenow={percentComplete}
      >
        <div className="hig__progress-bar__bar" style={this.fillBarStyles()}>
          <div className="hig__progress-bar__fill" />
          <svg
            width="3px"
            height="4px"
            viewBox="0 0 3 4"
            version="1.1"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <polygon
              id="end-right"
              points="0 0 2.68 0 1 4 0 4"
              fill="#0ED3BE"
            />
          </svg>
        </div>
      </div>
    );
  }
}
