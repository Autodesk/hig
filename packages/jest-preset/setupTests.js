require("jest-enzyme");
require("jest-extended");
require("./matchers");
