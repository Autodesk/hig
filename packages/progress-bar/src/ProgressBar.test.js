import { takeSnapshotsOf } from "@hig/jest-preset/helpers";
import ProgressBar from "./ProgressBar";

describe("progress-bar/ProgressBar", () => {
  describe("snapshot tests", () => {
    takeSnapshotsOf(ProgressBar, [
      {
        desc: "indeterminate"
      },
      {
        desc: "determinate",
        props: { percentComplete: 33 }
      },
      {
        desc: "passing className",
        props: { className: "myClass-a myClass-b" }
      },
      {
        desc: "custom stylesheet",
        props: {
          stylesheet: styles => ({
            ...styles,
            wrapper: {
              ...styles.wrapper,
              backgroundColor: "red"
            }
          })
        }
      }
    ]);
  });
});
