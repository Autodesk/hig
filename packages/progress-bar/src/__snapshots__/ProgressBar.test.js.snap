// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progress-bar/ProgressBar snapshot tests  1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(200px);
    -ms-transform: translateX(200px);
    transform: translateX(200px);
  }
}

.emotion-3 {
  position: relative;
  border-radius: 2px;
  background-color: rgba(0,0,0,0.05);
  overflow: hidden;
  width: 100%;
  height: 4px;
}

.emotion-2 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: -3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: width 0.3s ease-in-out;
  transition: width 0.3s ease-in-out;
  width: 100%;
  -webkit-animation: 2s linear infinite animation-0;
  animation: 2s linear infinite animation-0;
}

.emotion-0 {
  background-color: transparent;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
}

.emotion-0:before {
  content: "";
  position: absolute;
  right: 2px;
  width: 200px;
  height: 4px;
  background-image: linear-gradient(135deg,transparent,#38abdf 100%);
  background-size: 200px 4px;
}

.emotion-1 {
  fill: #38abdf;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  className="emotion-3"
  role="progressbar"
>
  <div
    className="emotion-2"
  >
    <div
      className="emotion-0"
    />
    <svg
      height="4px"
      version="1.1"
      viewBox="0 0 3 4"
      width="3px"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <polygon
        className="emotion-1"
        id="end-right"
        points="0 0 2.68 0 1 4 0 4"
      />
    </svg>
  </div>
</div>
`;

exports[`progress-bar/ProgressBar snapshot tests  2`] = `
.emotion-3 {
  position: relative;
  border-radius: 2px;
  background-color: rgba(0,0,0,0.05);
  overflow: hidden;
  width: 100%;
  height: 4px;
}

.emotion-1 {
  fill: #38abdf;
}

.emotion-2 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: -3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: width 0.3s ease-in-out;
  transition: width 0.3s ease-in-out;
  width: 33%;
}

.emotion-0 {
  background-color: #38abdf;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow={33}
  className="emotion-3"
  role="progressbar"
>
  <div
    className="emotion-2"
  >
    <div
      className="emotion-0"
    />
    <svg
      height="4px"
      version="1.1"
      viewBox="0 0 3 4"
      width="3px"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <polygon
        className="emotion-1"
        id="end-right"
        points="0 0 2.68 0 1 4 0 4"
      />
    </svg>
  </div>
</div>
`;

exports[`progress-bar/ProgressBar snapshot tests  3`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(200px);
    -ms-transform: translateX(200px);
    transform: translateX(200px);
  }
}

.emotion-3 {
  position: relative;
  border-radius: 2px;
  background-color: rgba(0,0,0,0.05);
  overflow: hidden;
  width: 100%;
  height: 4px;
}

.emotion-2 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: -3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: width 0.3s ease-in-out;
  transition: width 0.3s ease-in-out;
  width: 100%;
  -webkit-animation: 2s linear infinite animation-0;
  animation: 2s linear infinite animation-0;
}

.emotion-0 {
  background-color: transparent;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
}

.emotion-0:before {
  content: "";
  position: absolute;
  right: 2px;
  width: 200px;
  height: 4px;
  background-image: linear-gradient(135deg,transparent,#38abdf 100%);
  background-size: 200px 4px;
}

.emotion-1 {
  fill: #38abdf;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  className="emotion-3 myClass-a myClass-b"
  role="progressbar"
>
  <div
    className="emotion-2 myClass-a__progress-bar myClass-b__progress-bar"
  >
    <div
      className="emotion-0 myClass-a__fill myClass-b__fill"
    />
    <svg
      height="4px"
      version="1.1"
      viewBox="0 0 3 4"
      width="3px"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <polygon
        className="emotion-1 myClass-a__polygon myClass-b__polygon"
        id="end-right"
        points="0 0 2.68 0 1 4 0 4"
      />
    </svg>
  </div>
</div>
`;

exports[`progress-bar/ProgressBar snapshot tests  4`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(200px);
    -ms-transform: translateX(200px);
    transform: translateX(200px);
  }
}

.emotion-2 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: -3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: width 0.3s ease-in-out;
  transition: width 0.3s ease-in-out;
  width: 100%;
  -webkit-animation: 2s linear infinite animation-0;
  animation: 2s linear infinite animation-0;
}

.emotion-0 {
  background-color: transparent;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
}

.emotion-0:before {
  content: "";
  position: absolute;
  right: 2px;
  width: 200px;
  height: 4px;
  background-image: linear-gradient(135deg,transparent,#38abdf 100%);
  background-size: 200px 4px;
}

.emotion-1 {
  fill: #38abdf;
}

.emotion-3 {
  position: relative;
  border-radius: 2px;
  background-color: red;
  overflow: hidden;
  width: 100%;
  height: 4px;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  className="emotion-3"
  role="progressbar"
>
  <div
    className="emotion-2"
  >
    <div
      className="emotion-0"
    />
    <svg
      height="4px"
      version="1.1"
      viewBox="0 0 3 4"
      width="3px"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <polygon
        className="emotion-1"
        id="end-right"
        points="0 0 2.68 0 1 4 0 4"
      />
    </svg>
  </div>
</div>
`;

exports[`progress-bar/ProgressBar snapshot tests custom stylesheet 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(200px);
    -ms-transform: translateX(200px);
    transform: translateX(200px);
  }
}

.emotion-2 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: -3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: width 0.3s ease-in-out;
  transition: width 0.3s ease-in-out;
  width: 100%;
  -webkit-animation: 2s linear infinite animation-0;
  animation: 2s linear infinite animation-0;
}

.emotion-0 {
  background-color: transparent;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
}

.emotion-0:before {
  content: "";
  position: absolute;
  right: 2px;
  width: 200px;
  height: 4px;
  background-image: linear-gradient(135deg,transparent,#38abdf 100%);
  background-size: 200px 4px;
}

.emotion-1 {
  fill: #38abdf;
}

.emotion-3 {
  position: relative;
  border-radius: 2px;
  background-color: red;
  overflow: hidden;
  width: 100%;
  height: 4px;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  className="emotion-3"
  role="progressbar"
>
  <div
    className="emotion-2"
  >
    <div
      className="emotion-0"
    />
    <svg
      height="4px"
      version="1.1"
      viewBox="0 0 3 4"
      width="3px"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <polygon
        className="emotion-1"
        id="end-right"
        points="0 0 2.68 0 1 4 0 4"
      />
    </svg>
  </div>
</div>
`;

exports[`progress-bar/ProgressBar snapshot tests determinate 1`] = `
.emotion-3 {
  position: relative;
  border-radius: 2px;
  background-color: rgba(0,0,0,0.05);
  overflow: hidden;
  width: 100%;
  height: 4px;
}

.emotion-1 {
  fill: #38abdf;
}

.emotion-2 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: -3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: width 0.3s ease-in-out;
  transition: width 0.3s ease-in-out;
  width: 33%;
}

.emotion-0 {
  background-color: #38abdf;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow={33}
  className="emotion-3"
  role="progressbar"
>
  <div
    className="emotion-2"
  >
    <div
      className="emotion-0"
    />
    <svg
      height="4px"
      version="1.1"
      viewBox="0 0 3 4"
      width="3px"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <polygon
        className="emotion-1"
        id="end-right"
        points="0 0 2.68 0 1 4 0 4"
      />
    </svg>
  </div>
</div>
`;

exports[`progress-bar/ProgressBar snapshot tests indeterminate 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(200px);
    -ms-transform: translateX(200px);
    transform: translateX(200px);
  }
}

.emotion-3 {
  position: relative;
  border-radius: 2px;
  background-color: rgba(0,0,0,0.05);
  overflow: hidden;
  width: 100%;
  height: 4px;
}

.emotion-2 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: -3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: width 0.3s ease-in-out;
  transition: width 0.3s ease-in-out;
  width: 100%;
  -webkit-animation: 2s linear infinite animation-0;
  animation: 2s linear infinite animation-0;
}

.emotion-0 {
  background-color: transparent;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
}

.emotion-0:before {
  content: "";
  position: absolute;
  right: 2px;
  width: 200px;
  height: 4px;
  background-image: linear-gradient(135deg,transparent,#38abdf 100%);
  background-size: 200px 4px;
}

.emotion-1 {
  fill: #38abdf;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  className="emotion-3"
  role="progressbar"
>
  <div
    className="emotion-2"
  >
    <div
      className="emotion-0"
    />
    <svg
      height="4px"
      version="1.1"
      viewBox="0 0 3 4"
      width="3px"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <polygon
        className="emotion-1"
        id="end-right"
        points="0 0 2.68 0 1 4 0 4"
      />
    </svg>
  </div>
</div>
`;

exports[`progress-bar/ProgressBar snapshot tests passing className 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(200px);
    -ms-transform: translateX(200px);
    transform: translateX(200px);
  }
}

.emotion-3 {
  position: relative;
  border-radius: 2px;
  background-color: rgba(0,0,0,0.05);
  overflow: hidden;
  width: 100%;
  height: 4px;
}

.emotion-2 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: -3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-transition: width 0.3s ease-in-out;
  transition: width 0.3s ease-in-out;
  width: 100%;
  -webkit-animation: 2s linear infinite animation-0;
  animation: 2s linear infinite animation-0;
}

.emotion-0 {
  background-color: transparent;
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
}

.emotion-0:before {
  content: "";
  position: absolute;
  right: 2px;
  width: 200px;
  height: 4px;
  background-image: linear-gradient(135deg,transparent,#38abdf 100%);
  background-size: 200px 4px;
}

.emotion-1 {
  fill: #38abdf;
}

<div
  aria-valuemax="100"
  aria-valuemin="0"
  className="emotion-3 myClass-a myClass-b"
  role="progressbar"
>
  <div
    className="emotion-2 myClass-a__progress-bar myClass-b__progress-bar"
  >
    <div
      className="emotion-0 myClass-a__fill myClass-b__fill"
    />
    <svg
      height="4px"
      version="1.1"
      viewBox="0 0 3 4"
      width="3px"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <polygon
        className="emotion-1 myClass-a__polygon myClass-b__polygon"
        id="end-right"
        points="0 0 2.68 0 1 4 0 4"
      />
    </svg>
  </div>
</div>
`;
